cmd/dns-operator/hosts_updater.go

```diff
@@ -96,36 +114,26 @@ func (s *HostsUpdater) sync() error {
 			entries = append(entries, fmt.Sprintf("%s %s %s # %s", ip.String(), svc, fqdn, marker))
 		}
 	}
+	for _, entry := range entries {
+		newContent += entry + "\n"
+	}
 
-	// Read the existing hosts file, excluding lines previously generated
-	// from the hosts updater, so we can re-generate our entries.
-	file, err := ioutil.ReadFile(s.HostsFile)
+	// Write a temporary hosts file with the updated contents.
+	newHostsFile, err := ioutil.TempFile("", "hosts")
```

Comment:
Need to copy the old file's attributes (owener, permissions, and SELinux context).

Reply:
Good catch...